cmd/main.go

```diff
@@ -129,6 +132,16 @@ func main() {
 		providers = defaultProviders
 	}
 
+	// Display version here.
+
+	dnsInfoDisplay := fmt.Sprintf("dns-operator %s (%s-dirty)", version, commit)
```

Comment:
after setting the log options in the logger, I would implement something like the limitador operator is doing. 

```go
func printControllerMetaInfo() {                                                        
    setupLog := log.Log                                                                 
                                                                                        
    setupLog.Info(fmt.Sprintf("go version: %s", runtime.Version()))                     
    setupLog.Info(fmt.Sprintf("go os/arch: %s/%s", runtime.GOOS, runtime.GOARCH))       
    setupLog.Info("base logger", "log level", logLevel, "log mode", logMode)            
    setupLog.Info("", "version", version.Version, "commit", gitSHA, "dirty", dirty)     
}                                                                                       

```

this function should be called first thing in `main` ideally. In the limitador operator the logger is set at `init` time, even before `main` is called because it is based on env vars. If this the DNS operator relies on command line args to configure the logger, it needs to be done in main. So, first thing to do is to  parse command line args, then set the logger and then, inmediately log operator version and other info. If there is any error afterwards, we can tell which version of the operator is being run.